src/app/shared/modal/modal-login/modal-login.component.ts

```diff
@@ -35,14 +35,23 @@ export class ModalLoginComponent implements OnInit {
   }
 
   login(): void {
-    this.courseService.loadingSubject.next(true);
-    this.authService.login(this.email, this.password);
+    this.loadingService.loadingSubject.next(true);
+    this.authService.login(this.email, this.password).subscribe( 
+      (response) => {
+        const token = JSON.parse(JSON.stringify(response)).token;
+        localStorage.setItem("token", token);
+        this.authService.getUserLogin().subscribe(response => {
+          this.authService.userSubject.next(JSON.parse(JSON.stringify(response)));
+        })
+        this.authService.isUserLogged.next(true);
+        this.router.navigate(['/courses']);
```

Comment:
Try to move logic to the service using the **RxJs** operators
You can also move the error handling to the operators
Much logic is always better to handle in the service

Reply:
ok, removed logic to the service